test/Integration/AddProductsNoPartialUpsert.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Azure.WebJobs.Extensions.Sql.Samples.Common;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Sql.Tests.Integration
+{
+    public static class AddProductsNoPartialUpsert
+    {
+        // This output binding should throw an error since the ProductsNameNotNull table does not 
+        // allows rows without a Name value. No rows should be upserted to the Sql table.
+        [FunctionName("AddProductsNoPartialUpsert")]
+        public static IActionResult Run(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "addproducts-nopartialupsert")]
+            HttpRequest req,
+            [Sql("dbo.ProductsNameNotNull", ConnectionStringSetting = "SqlConnectionString")] ICollector<Product> products)
+        {
+            List<Product> newProducts = ProductUtilities.GetNewProducts(1000);
```

Comment:
It would be nice to use the batch size as a constant in case we change that later